IndicoV2/Workflows/WorkflowsClient.cs

```diff
@@ -1,34 +1,45 @@
 ﻿using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Linq;
 using IndicoV2.StrawberryShake;
-using IndicoV2.V1Adapters.Workflows;
 using IndicoV2.Workflows.Models;
 
 namespace IndicoV2.Workflows
 {
     public class WorkflowsClient : IWorkflowsClient
     {
-        private readonly WorkflowsV1ClientAdapter _legacy;
         private readonly IndicoStrawberryShakeClient _strawberryShake;
+        private readonly IndicoClient _indicoClient;
 
         public WorkflowsClient(IndicoClient indicoClient)
         {
-            _legacy = new WorkflowsV1ClientAdapter(indicoClient.LegacyClient);
+            _indicoClient = indicoClient;
             _strawberryShake = indicoClient.IndicoStrawberryShakeClient;
         }
 
-        public Task<IEnumerable<IWorkflow>> ListAsync(int dataSetId, CancellationToken cancellationToken = default) =>
-            _legacy.ListAsync(dataSetId, cancellationToken);
+        public async Task<IEnumerable<IWorkflow>> ListAsync(int dataSetId, CancellationToken cancellationToken = default) 
+        {
+            var result = await _strawberryShake.Workflows().ListAsync(dataSetId, cancellationToken);
+            return result.Workflows.Select(x => ToWorkflow(x)).ToList();
+        }
 
-        public Task<IEnumerable<IWorkflow>>
-            ListAsync(int[] dataSetIds, CancellationToken cancellationToken = default) =>
-            _legacy.ListAsync(dataSetIds, cancellationToken);
+        public async Task<IEnumerable<IWorkflow>>ListAsync(int[] dataSetIds, CancellationToken cancellationToken = default)
+        {
+            var result = await _strawberryShake.Workflows().ListAsync(dataSetIds, cancellationToken);
+            return result.Workflows.Select(x => ToWorkflow(x)).ToList();
+        }
 
         public Task<IWorkflowAddDataResult> AddDataAsync(int workflowId, CancellationToken cancellationToken) =>
             _strawberryShake.Workflows().AddData(workflowId, cancellationToken);
 
         public Task<WorkflowStatus> GetStatusAsync(int workflowId, CancellationToken cancellationToken) =>
             _strawberryShake.Workflows().GetStatus(workflowId, cancellationToken);
+
+        private Workflow ToWorkflow(IListWorkflows_Workflows_Workflows workflow) => new Workflow {
```

Comment:
@goatrocks I made the changes you suggested, and the compiler is complaining. We can't change the type of the properties of the Workflow class, because it implements the IWorkflow interface. I was hoping providing a default value for id and revierEnabled would allow us to pass in potentially null values, but apparently not :/

Reply:
we will have to check for null and pass in a default when converting. you can coalesce values like workflow.id ?? 0 when using the property constructor :)